package/src/components/MessageList/MessageList.tsx

```diff
@@ -783,28 +785,46 @@ const MessageListWithContext = <
     }
   };
 
+  const scrollToIndexFailedRetryCountRef = useRef<number>(0);
   const onScrollToIndexFailedRef = useRef<
     FlatListProps<MessageType<StreamChatGenerics>>['onScrollToIndexFailed']
-  >((info) => {
+  >(async (info) => {
     // We got a failure as we tried to scroll to an item that was outside the render length
-    if (flatListRef.current) {
-      // we dont know the actual size of all items but we can see the average, so scroll to the closest offset
-      flatListRef.current.scrollToOffset({
+    if (!flatListRef.current) return;
+    // we don't know the actual size of all items but we can see the average, so scroll to the closest offset
+    flatListRef.current.scrollToOffset({
+      animated: false,
+      offset: info.averageItemLength * info.index,
+    });
+    // since we used only an average offset... we won't go to the center of the item yet
+    // with a little delay to wait for scroll to offset to complete, we can then scroll to the index
+    await new Promise((resolve) => setTimeout(resolve, WAIT_FOR_SCROLL_TO_OFFSET_TIMEOUT));
+
+    // Only when index is greater than 0 and in range of items in FlatList
+    // this onScrollToIndexFailed will be called again
+    try {
+      flatListRef.current?.scrollToIndex({
         animated: false,
-        offset: info.averageItemLength * info.index,
+        index: info.index,
+        viewPosition: 0.5, // try to place message in the center of the screen
       });
-      // since we used only an average offset.. we wont go to the center of the item yet
-      // with a little delay to wait for scroll to offset to complete, we can then scroll to the index
-      setTimeout(() => {
-        flatListRef.current?.scrollToIndex({
-          animated: false,
-          index: info.index,
-          viewPosition: 0.5, // try to place message in the center of the screen
-        });
-        if (messageIdLastScrolledToRef.current) {
-          setTargetedMessage(messageIdLastScrolledToRef.current);
-        }
-      }, 150);
+
+      if (messageIdLastScrolledToRef.current) {
+        setTargetedMessage(messageIdLastScrolledToRef.current);
+      }
+    } catch (e) {
+      if (
+        !onScrollToIndexFailedRef.current ||
+        scrollToIndexFailedRetryCountRef.current > MAX_RETRIES_AFTER_SCROLL_FAILURE
+      ) {
+        throw e;
```

Comment:
my view is that we shouldnt throw here..

lets just gracefully log the error

The error we may get is index not found in the array.. which means we cant scroll to it..

we dont need to throw an error for that.. it may be uncatched the integrator